src/core/website_manager.py

```diff
@@ -1,311 +1,218 @@
+                return os.path.splitext(urlparse(link)[2])[-1]
+
+            file_extensions = [extension_from_link(link) for link in raw_links]
+            return [x for x in get_unique_list(file_extensions) if x != ""]
+
+        def preprocess_header(header: str) -> dict:
+            header = header.lower()
+
+            idx = max(header.find('b"'), header.find("b'"))
+            if idx >= 0:
+                header = (
+                    header.replace("b'", '"')
+                    .replace('b"', '"')
+                    .replace("/'", '"')
+                    .replace("'", '"')
+                    .replace('""', '"')
+                    .replace('/"', "/")
+                )
 
-        attributes = [
-            "href",
-            "src",
-            "srcset",
-            "img src",
-            "data-src",
-            "data-srcset",
-        ]
-        links = [
-            element.attrs.get(attribute)
-            for tag in unique_tags
-            for element in get_unique_list(
-                self.website_data.soup.find_all(tag)
+            if len(header) > 0:
+                return json.loads(header)
+            return {}
+
+        fetched_from_splash = False
+        if (
+            message.html is None
+            or message.header is None
+            or message.har is None
+        ):
+            logger.debug(
+                f"Missing one of html, header, or har -> fetching {message.url}"
             )
-            for attribute in attributes
-            if element.has_attr(attribute)
-        ]
-        self._logger.debug(
-            f"Found links at {time.perf_counter() - global_start} since start: {len(links)}"
-        )
-        self.website_data.raw_links = get_unique_list(
-            links
-            + [el for el in self.website_data.image_links if el is not None]
-            + script_links
+            message = cls.fetch_content(message)
+            fetched_from_splash = True
+
+        def headers_from_splash(
+            headers: list[dict[str, str]]
+        ) -> dict[str, str]:
+            """
+            Splash returns headers as a list of dicts with 'name' and 'value' entries for each header field.
+            Here this list of dicts is transformed into a single dictionary with lowercase header field names.
+
+            Note: If splash does not normalize duplicate header fields into a comma separate list, then we will lose
+            them here!
+            """
+            return {d["name"].lower().strip(): d["value"] for d in headers}
+
+        soup = BeautifulSoup(message.html, "lxml")
+        image_links = [image.attrs.get("src") for image in soup.findAll("img")]
+        raw_links = extract_raw_links(soup=soup, image_links=image_links)
+        return WebsiteData(
+            url=message.url,
+            html=message.html,
+            # fixme: remove the separate code paths
+            raw_header=json.dumps(message.header)
```

Comment:
Here the same as in the other file - this multiline if-else confuses me. Please place it before expressing it here.

Reply:
done